src/main/java/hudson/plugins/active_directory/ActiveDirectoryUnixAuthenticationProvider.java

```diff
@@ -461,8 +461,16 @@ private String getPrincipalName(String username, String domainName) {
                             throw e;
                         }
                     }
-                    if (!found || duration >= 10) {
-                        LOGGER.warning(String.format("AD chain lookup is taking too long (%dms). Falling back to recursive lookup", duration));
+                    if (!found && duration >= 10) {
+                        LOGGER.warning(String.format("AD chain didn't find the group in (%dms). Falling back to recursive lookup", duration));
+                        groupLookupStrategy = GroupLookupStrategy.RECURSIVE;
+                        continue;
+                    } else if (found && duration >= 10) {
+                        LOGGER.warning(String.format("AD chain lookup found the group but took too long (%dms). Falling back to recursive lookup", duration));
```

Comment:
duration is in seconds not milliseconds.
This is the wrong way to do [parameterized logging](https://docs.oracle.com/javase/7/docs/api/java/util/logging/Logger.html#log(java.util.logging.Level,%20java.lang.String,%20java.lang.Object))